thrift/src/main/java/com/intellij/plugins/thrift/editor/GoToThriftDefinition.java

```diff
@@ -26,60 +26,67 @@
 import java.time.Duration;
 import java.util.List;
 
+import javax.swing.Icon;
+
 public class GoToThriftDefinition extends AnAction {
   static Logger logger = Logger.getInstance(GoToThriftDefinition.class);
 
   @Override
   public void actionPerformed(@NotNull AnActionEvent event) {
     PsiElement psiElement = event.getData(CommonDataKeys.PSI_ELEMENT);
     List<ThriftDeclaration> thriftMatches = getThriftDeclarations(psiElement, event.getProject());
-    if (thriftMatches.size() > 0) {
-      PsiElementListCellRenderer<PsiElement> renderer = new PsiElementListCellRenderer<PsiElement>() {
-        @Override
-        public String getElementText(PsiElement element) {
-          if (element instanceof ThriftStruct) {
-            return ((ThriftStruct) element).getName();
-          } else {
-            return element.getText().substring(0, 20);
-          }
-        }
+    if (thriftMatches.size() == 1) {
+      // navigate directly if there's only 1 matched declaration
+      thriftMatches.get(0).navigate(true);
```

Comment:
Navigate directly if `thriftMatches` has only 1 element since there's no similar method in `PsiTargetNavigator` to handle both navigation/creating popup like `PsiElementListNavigator.navigateOrCreatePopup`